### PR TITLE
#28989: Fix CID 1442277

### PR DESCRIPTION
**Pull request by rl1987**
https://trac.torproject.org/projects/tor/ticket/28989

### CODE REVIEW

**Comment by nmathewson** [comment]
Should the helper be called in the "done" block, maybe?  Remember that every tt_* macro will conditionally call "goto done" on failure, so doing cleanup at other points in the function is not necessarily useful.

File: src/test/test_hs_service.c [hunk 1]
```diff
@@ -289,6 +289,20 @@ helper_create_service(void)
   return service;
 }
 
+/* Helper: Deallocate a given service object, it's child objects and
```
Comment (nmathewson) [inline src/test/test_hs_service.c]: s/it's/its/

Comment (rl1987) [reply src/test/test_hs_service.c]: Fixed - 151ce23dc22383ebc754931e288bc2ad604cece9.

**Comment by rl1987** [comment]
In 0f0ad3c2530c62d94b339f6c752effa2d80c8443 I keep track of the last `hs_service_t` struct that was created in one of these inner lexical scopes and deallocate it in case one of the asserts failed. I suppose this lets Coverity realise that we deallocate it either way?

**Comment by nmathewson** [comment]
squashed and merged

**Pull request closed without merging**